stdlib/SuiteSparse/src/cholmod.jl

```diff
@@ -460,157 +455,132 @@ function norm_dense(D::Dense{Tv}, p::Integer) where Tv<:VTypes
                 ptr, common_struct)
 end
 
 function aat(A::Sparse{Tv}, fset::Vector{SuiteSparse_long}, mode::Integer) where Tv<:VRealTypes
-    Sparse(ccall((@cholmod_name("aat", SuiteSparse_long), :libcholmod),
+    Sparse(ccall((@cholmod_name("aat"), :libcholmod),
         Ptr{C_Sparse{Tv}},
             (Ptr{C_Sparse{Tv}}, Ptr{SuiteSparse_long}, Csize_t, Cint, Ptr{UInt8}),
                 A, fset, length(fset), mode, common_struct))
 end
 
 function sparse_to_dense(A::Sparse{Tv}) where Tv<:VTypes
-    Dense(ccall((@cholmod_name("sparse_to_dense", SuiteSparse_long),:libcholmod),
+    Dense(ccall((@cholmod_name("sparse_to_dense"),:libcholmod),
         Ptr{C_Dense{Tv}},
             (Ptr{C_Sparse{Tv}}, Ptr{UInt8}),
                 A, common_struct))
 end
 function dense_to_sparse(D::Dense{Tv}, ::Type{SuiteSparse_long}) where Tv<:VTypes
-    Sparse(ccall((@cholmod_name("dense_to_sparse", SuiteSparse_long),:libcholmod),
+    Sparse(ccall((@cholmod_name("dense_to_sparse"),:libcholmod),
         Ptr{C_Sparse{Tv}},
             (Ptr{C_Dense{Tv}}, Cint, Ptr{UInt8}),
                 D, true, common_struct))
 end
 
 function factor_to_sparse!(F::Factor{Tv}) where Tv<:VTypes
     ss = unsafe_load(pointer(F))
-    ss.xtype > PATTERN || throw(CHOLMODException("only numeric factors are supported"))
-    Sparse(ccall((@cholmod_name("factor_to_sparse", SuiteSparse_long),:libcholmod),
+    ss.xtype == PATTERN && throw(CHOLMODException("only numeric factors are supported"))
+    Sparse(ccall((@cholmod_name("factor_to_sparse"),:libcholmod),
         Ptr{C_Sparse{Tv}},
             (Ptr{C_Factor{Tv}}, Ptr{UInt8}),
                 F, common_struct))
 end
 
-function change_factor!(::Type{Float64}, to_ll::Bool,
-        to_super::Bool, to_packed::Bool, to_monotonic::Bool, F::Factor{Tv}) where Tv<:VTypes
-    @isok ccall((@cholmod_name("change_factor", SuiteSparse_long),:libcholmod), Cint,
+function change_factor!(F::Factor{Tv}, to_ll::Bool, to_super::Bool, to_packed::Bool,
+                        to_monotonic::Bool) where Tv<:VTypes
+    @isok ccall((@cholmod_name("change_factor"),:libcholmod), Cint,
             (Cint, Cint, Cint, Cint, Cint, Ptr{C_Factor{Tv}}, Ptr{UInt8}),
-                REAL, to_ll, to_super, to_packed, to_monotonic, F, common_struct)
-    # don't register finalizer since we reuse object
-    Factor{Float64}(pointer(F), false)
-end
-
-function change_factor!(::Type{Complex{Float64}}, to_ll::Bool,
-        to_super::Bool, to_packed::Bool, to_monotonic::Bool, F::Factor{Tv}) where Tv<:VTypes
-    @isok ccall((@cholmod_name("change_factor", SuiteSparse_long),:libcholmod), Cint,
-            (Cint, Cint, Cint, Cint, Cint, Ptr{C_Factor{Tv}}, Ptr{UInt8}),
-                COMPLEX, to_ll, to_super, to_packed, to_monotonic, F, common_struct)
-    # don't register finalizer since we reuse object
-    Factor{Complex{Float64}}(pointer(F), false)
+                xtyp(Tv), to_ll, to_super, to_packed, to_monotonic, F, common_struct)
```

Comment:
Is it on purpose that you don't return the `Factor` anymore?

Reply:
Yes. (1) The return value wasn't used in any code, and (2) if it was used it was likely to cause a segfault, since there was nothing to stop the original object from being GC'ed.